Add global environment variables

surminus opened the pull request:
This commit creates an "env_file" which is used by each container that is started using the govuk_containers::app defined type. It separates the management of app specific env vars and global env vars rather than trying to merge arrays of both into the same variable.

On testing they seem to correctly get loaded (using `docker inspect` to check):

  ```
  vagrant@docker-backend-1:~$ sudo docker inspect release |grep -A10 Env
            "Env": [
                "GOVUK_ENV=development",
                "NODE_ENV=development",
                "RACK_ENV=development",
                "RAILS_ENV=development",
                "ERRBIT_ENVIRONMENT_NAME=vagrant",
                "GOVUK_APP_DOMAIN=dev.gov.uk",
                "GOVUK_ASSET_HOST=http://static.dev.gov.uk",
                "GOVUK_ASSET_ROOT=http://static.dev.gov.uk",
                "GOVUK_WEBSITE_ROOT=http://www.dev.gov.uk",
                "FOO=BAR",
  ```

afda16 commented on hieradata/vagrant.yaml:
```diff
@@ -29,19 +29,19 @@ govuk::apps::stagecraft::worker::enabled: true
+  - "FOO=BAR"
+
+govuk_containers::app::config::global_envvars:
+  - "GOVUK_ENV=development"
```
I think govuk_env defaults to 'production' in the deploy::config class, and we only override with 'development' in:
`hieradata/development.yaml:govuk::deploy::config::govuk_env: 'development'`

afda16 commented on modules/govuk_containers/manifests/app.pp:
```diff
@@ -17,6 +17,10 @@
 # [*envvars*]
 #   An array of environment variables to start the container with.
 #
+# [*global_envs*]
```
minor update to global_env_file

afda16 commented:
I think a bin file has gone in the commit:
modules/govuk_containers/manifests/app/.config.pp.swp

Where are now the ...::release::envvars configured?

surminus commented:
I've removed the bin file, thanks! Addressed comment.

The app env vars are configured in this PR: https://github.com/alphagov/govuk-puppet/pull/5837
